### DISCUSSION TITLE
Ubuntu 18.04 build error

### DISCUSSION THREAD

**Discussion by KuiWei004** [google/xls#349] [Q&A]
Hi all,

my linux evironment is ubuntu 18.04, and for building xls, I update gcc/g++ to gcc/g++ 10, but when I build xls, I  met the following errors:

src/main/tools/linux-sandbox-pid1.cc:390: "execvp(/usr/local/bin/gcc, 0x17f03b0)": No such file or directory

how can I fix that?

Please help.

**Comment by cdleary** [comment]
Hi @KuiWei004 sorry for the delayed response! Just want to note that supporting various different build environments is kind of "we'll try not to break them intentionally but if it's not in CI it's difficult / patches welcome" for XLS at this time, but I'm hopeful we can get both 20.04 and 18.04 to work since they're not too different. I was able to land some of the changes that enabled 18.04 from google/xls#159, but I think this tweak:

https://github.com/google/xls/issues/159#issuecomment-716061539

To do with `_IO_ftrylockfile` is still necessary, I need to figure out a better way to address that which is compatible with both 20.04 and 18.04. Will try to look at how to do that, but I believe the workaround is the commenting out, for now.

Will continue trying to close out #159 -- if we can find a solution I think we can add 18.04 to CI to try to prevent regressions.

Also note that if you're seeing particular environment issues you can always try the docker command that runs with 18.04 in the repository to see if a "clean" environment reproduces / see the commands docker is using: `docker build . -f Dockerfile-ubuntu-18.04`

**Reply by cdleary** [reply]
Gave an update in https://github.com/google/xls/issues/159#issuecomment-809641412 -- I think it could be fixed now, let's discuss any further issues there, thanks!

**Reply by RohitPatil555** [reply]
I verified on ubuntu 20.04. Build is working.

**Comment by proppy** [comment]
closing per https://github.com/google/xls/discussions/349#discussioncomment-746501
